scripts/dev.sh

```diff
@@ -37,31 +37,66 @@ function docker_network_remove {
   docker network remove "$1"
 }
 
-trap onexit EXIT
-
-if ! docker_network_exist "$MESG_NAME"; then
-  docker_network_create "$MESG_NAME"
-fi
-
-if ! docker_network_exist "$MESG_TENDERMINT_NETWORK"; then
-  docker_network_create "$MESG_TENDERMINT_NETWORK"
-fi
-
-mkdir -p $MESG_PATH
-
-echo "create docker service: "
-docker service create \
-  --name $MESG_NAME \
-  --tty \
-  --label com.docker.stack.namespace=$MESG_NAME \
-  --label com.docker.stack.image=mesg/engine:local \
-  --env MESG_NAME=$MESG_NAME \
-  --mount type=bind,source=/var/run/docker.sock,destination=/var/run/docker.sock \
-  --mount type=bind,source=$MESG_PATH,destination=/root/.mesg \
-  --network $MESG_NAME \
-  --network name=$MESG_TENDERMINT_NETWORK \
-  --publish $MESG_SERVER_PORT:50052 \
-  --publish $MESG_TENDERMINT_PORT:26656 \
-  mesg/engine:local
-
-docker service logs --follow --raw $MESG_NAME
+function start_engine {
+  if ! docker_network_exist "$MESG_NAME"; then
+    docker_network_create "$MESG_NAME"
+  fi
+
+  if ! docker_network_exist "$MESG_TENDERMINT_NETWORK"; then
+    docker_network_create "$MESG_TENDERMINT_NETWORK"
+  fi
+
+  mkdir -p $MESG_PATH
+
+  echo "create docker service: "
+  docker service create \
+    --name $MESG_NAME \
+    --tty \
+    --label com.docker.stack.namespace=$MESG_NAME \
+    --label com.docker.stack.image=mesg/engine:local \
+    --env MESG_NAME=$MESG_NAME \
+    --mount type=bind,source=/var/run/docker.sock,destination=/var/run/docker.sock \
+    --mount type=bind,source=$MESG_PATH,destination=/root/.mesg \
+    --network $MESG_NAME \
+    --network name=$MESG_TENDERMINT_NETWORK \
+    --publish $MESG_SERVER_PORT:50052 \
+    --publish $MESG_TENDERMINT_PORT:26656 \
+    mesg/engine:local
+}
+
+function stop_engine {
+  onexit
+}
+
+quiet=false
+
+while getopts "q" o; do
+  case $o in
+    q)
+      quiet=true
+      ;;
+    *)
+      echo "unknown flag $0"
+      exit 1
+      ;;
+  esac
+done
+shift $((OPTIND-1))
```

Comment:
what?

Reply:
It's bash default syntax to remove element from array. In that case it removes element from arguments array.